progress/README.md

```diff
@@ -0,0 +1,256 @@
+  - **Code:** 200
+    - **Content:** 
+    ```json
+		{
+      "message": "User progress retrieved successfully.",
+      "progress": [
+        { "<user_progress_object_1>" },
+        { "<user_progress_object_2>" },
+        ..
+      ]
+    }
+    ```
+- **Error Response:**
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+- **Example:**
+  GET /progress?type=task
+    ```json
+    {
+			"message": "All User Progress found successfully.",
+			"progress": [
+				{
+				"id": "progress1",
+				"type": "task",
+				"userId": "b0ad77cf461633s7745d59d8",
+				"taskId": "d576s3cd4450df7698731b7a",
+				"completed": "added modal progress",
+				"planned": "write test for modal progress",
+				"blockers": "None",
+				"createdAt": 1654101900000
+				},
+				{
+				"id": "progress2",
+				"type": "user",
+				"userId": "4d973377d1ad504b8cs6f576",
+				"completed": "added controllers",
+				"planned": "add model functions",
+				"blockers": "Pending review",
+				"date": "2023-04-30",
+				"createdAt": 1654101900400
+				}
+			]
+    }
+    ```
+	GET /progress?type=user&id=user123
+	```json
+	{
+		"message": "User progress retrieved successfully.",
+		"progress": [
+			{
+				"id": "user123",
+				"type": "user",
+				"userId": "user123",
+				"completed": "made backend changes for the feature a23",
+				"planned": "make frontend changes for the feature a23",
+				"blockers": "Pending review",
+				"date": "2023-04-30",
+				"createdAt": "2023-04-30T12:00:00Z"
+			}
+		]
+	}
+
+	```
+
+
+
+## **POST /progress**
+
+	Creates a new progress entry.
+- **Params**  
+  None
+- **Query**
+  None
+- **Body**  
+  Attributes:
+    -type (required, string): Specifies the type of the progress entry (e.g., "task", "user").
+    -userId (required, string): User ID associated with the progress entry.
+    -taskId (optional, string): Task ID associated with the progress entry (applicable for task progress only).
+    -completed (required, string): Completed portion of the task or progress.
+    -planned (required, string): Planned portion of the task or progress.
+    -blockers (required, string): Blockers for the progress entry.
+    -date (required, timestamp): Date of the progress entry (applicable for user progress only).
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** 
+    ```json
+		{
+			"message": "Progress entry created successfully.",
+			"progress": { "<progress_object>" }
+		}
+    ```
+- **Error Response:**
+  - **Code:** 401
+    - **Content:** 
+		```json
+		{
+			"statusCode": 401,
+			"error": "Unauthorized",
+			"message": "Authentication credentials are missing or invalid."
+		}
+		```
+  - **Code:** 404
+    - **Content:** 
+  	```json
+		{
+			"statusCode": 404,
+			"error": "Not Found",
+			"message": "Task or user not found"
+		}
+		```
```

Comment:
would we get a 404 for a post? in case task id is wrong shouldn't it send a bad request response?